### DISCUSSION TITLE
posts/Go/go_rabbitmq_tutorials_02/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#326] [Announcements]
https://www.liwenzhou.com/posts/Go/go_rabbitmq_tutorials_02/ 

 李文周的Blog rabbitmq 消息队列 任务队列 Go语言 客户端 教程 amqp

**Comment by itgcl** [comment]
binding key 和 routing key怎么区分的

**Comment by itgcl** [comment]
binding key 和 routing key怎么区分的

**Comment by callmePicacho** [comment]
> @itgcl
> binding key 和 routing key怎么区分的

感谢你的问题，带着你的问题看到后面明白了，回来给后来人解释一下：
- routing key 是消息发给交换器的 key
- binding key 是交换器和队列绑定 key
当交换器类型为 direct 或 topic 时，根据 routing key 将消息发送到能成功匹配上 binding key 规则的队列中
